rules-tests/Php80/Rector/Switch_/ChangeSwitchToMatchRector/Fixture/default_outside_switch.php.inc

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Rector\Tests\Php80\Rector\Switch_\ChangeSwitchToMatchRector\Fixture;
+
+final class DefaultOutsideSwitch
+{
+    public function run($input)
+    {
+        $value = 0;
+        switch ($input) {
+            case 100:
+                $value = 1000;
+                break;
+            case 200:
+                $value = 2000;
+                break;
+        }
+
+        return $value;
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\Php80\Rector\Switch_\ChangeSwitchToMatchRector\Fixture;
+
+final class DefaultOutsideSwitch
+{
+    public function run($input)
+    {
+        $value = match ($input) {
+            100 => 1000,
+            200 => 2000,
+            default => 0,
+        };
```

Comment:
I think the initialization `$value` should not be deleted, and then use `default => $value` as it initialized ref https://3v4l.org/tQYfq

```suggestion
        $value = 0;
        $value = match ($input) {
            100 => 1000,
            200 => 2000,
            default => $value,
        };
```

Before Assign removal, you may need to check if the default value use the assign in 

https://github.com/rectorphp/rector-src/blob/3b0e8b64866125b42f2d06fd84715c6263e41a12/rules/Php80/Rector/Switch_/ChangeSwitchToMatchRector.php#L144-L146

then verify to avoid remove next node:

https://github.com/rectorphp/rector-src/blob/3b0e8b64866125b42f2d06fd84715c6263e41a12/rules/Php80/Rector/Switch_/ChangeSwitchToMatchRector.php#L187

Reply:
Another alternative is change to: 

```php
    $value = 0;
    return match ($input) {
        100 => 1000,
        200 => 2000,
        default => $value,
    };
```

or 

```php
    return match ($input) {
        100 => 1000,
        200 => 2000,
        default => 0,
    };
```

Reply:
Thanks @samsonasik, I figured returning the match directly might be outside the scope of this Rector, and would probably be quite complex to make that only happen when `$value` is never accessed again before returning.

Keeping the initialization of `$value` would probably be the easiest solution and make the most sense with the scope of this Rector.